lms/djangoapps/courseware/views/views.py

```diff
@@ -1666,6 +1703,37 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True):
 
         missed_deadlines, missed_gated_content = dates_banner_should_display(course_key, request.user)
 
+        # Some content gating happens only at the Sequence level (e.g. "has this
+        # timed exam started?").
+        ancestor_seq = enclosing_sequence_for_gating_checks(block)
+        if ancestor_seq:
+            seq_usage_key = ancestor_seq.location
+            # We have a Descriptor, but I had trouble getting a SequenceModule
+            # from it (even using ._xmodule to force the conversion) because the
+            # runtime wasn't properly initialized. This view uses multiple
+            # runtimes (including Blockstore), so I'm pulling it from scratch
+            # based on the usage_key. We'll have to watch the performance impact
+            # of this. :(
+            seq_module_descriptor, _ = get_module_by_usage_id(
+                request, str(course_key), str(seq_usage_key), disable_staff_debug_info=True, course=course
+            )
+
+            # I'm not at all clear why get_module_by_usage_id returns the
+            # descriptor or why I need to manually force it to load the module
+            # like this, but I do... Hopefully this weirdness goes away when
+            # SequenceModule gets converted to an XBlock.
+            seq_module = seq_module_descriptor._xmodule
```

Comment:
Oof, bummer! That is wacky. Glad you were able to find a workaround, though.

Consider mentioning BD-04 and/or linking to https://github.com/edx/edx-platform/pull/25965 in the comment. Hopefully we can remove this in the near future.